src/Mapster/Utils/ReflectionUtils.cs

```diff
@@ -37,6 +38,24 @@ public static IEnumerable<IMemberModel> GetPublicFieldsAndProperties(this Type t
             return properties.Concat(fields);
         }
 
+        public static IEnumerable<IMemberModel> GetPrivateFieldsAndProperties(this Type type, bool allowNoSetter = true)
+        {
+            var properties = type.GetProperties(BindingFlags.Instance | BindingFlags.NonPublic)
+                .Where(x => allowNoSetter || x.CanWrite)
+                .Select(CreateModel);
+
+            var fields = type.GetFields(BindingFlags.Instance | BindingFlags.NonPublic)
+                .Where(x => (allowNoSetter || !x.IsInitOnly) && !IsBackingField(x))
+                .Select(CreateModel);
+
+            return properties.Concat(fields);
+        }
+
+        private static bool IsBackingField(FieldInfo field)
+        {
+            return field.CustomAttributes.Any(a => a.AttributeType == typeof(CompilerGeneratedAttribute));
+        }
```

Comment:
could you elaborate a bit more on `IgnoreAttribute`? I'm not aware of any .NET attribute with the name `IgnoreAttribute`, unless it's from NUnit. Or do you mean we create a custom attribute to ignore some private fields? The reason I don't want to include backing field is to avoid duplication, since we already get normal properties which use those backing fields